### PR TITLE
ENH add benchopt publish command

### PR DESCRIPTION
**Pull request by tomMoral**
Add a `benchopt publish` command to upload the benchmark results to [`benchopt/results`](https://github.com/benchopt/results).

The command is based on `pygithub` and upload the file when given a token. The token can be set either directly with `-t TOKEN`, with an environment variable `BENCHO_GITHUB_TOKEN` or in the config file.

### CODE REVIEW

File: benchopt/cli.py [hunk 1]
```diff
@@ -160,29 +161,47 @@ def run(benchmark, solver_names, forced_solvers, dataset_names,
...
-                f"Could not find result file {filename}. Available result "
-                f"files are:\n- {all_csv_files}"
-            )
-    else:
-        result_filename = all_csv_files[-1]
+    # Get the result file
+    benchmark = Benchmark(benchmark)
+    result_filename = benchmark.get_result_file(filename)
 
+    # Plot he results.
```
Comment (tomMoral) [inline benchopt/cli.py:168]:
```suggestion
    # Plot the results.
```

**Review by tomMoral — COMMENTED** [review]
@agramfort do you want to review this one?

**Review by agramfort — COMMENTED** [review]
works like a charm !

shall we push some doc in this PR?

**Comment by tomMoral** [comment]
Merging this as it fixes some errors for plotting. Let's do some Doc in a follow up PR.

**Comment by zaccharieramzi** [comment]
Is it possible that this PR has introduced a behaviour where plots are now presented in separate figures (and saved as such)?
If so is this behaviour intended?

**Pull request merged**